python/bind_data_access.h

```diff
@@ -389,12 +389,12 @@ void bind_data_properties(pybind11::class_<T, Ignored...> &c) {
   c.def_property_readonly("shape",
                           [](const T &self) {
                             const auto &dims = self.dims();
-                            auto shape{std::vector<scipp::index>(
-                                dims.shape().begin(), dims.shape().end())};
+                            py::list list = py::cast(dims.shape());
+
                             if (dims.sparse()) {
-                              shape.emplace_back(Dimensions::Sparse);
+                              list.append(py::none());
                             }
-                            return shape;
+                            return list;
```

Comment:
Can we add a unit test for this?

@nvaytet Does this change have any effect on the plotting code?

Reply:
Just to note that now both Variables and Dataset will return None on the sparse dim's shape